R/codecov.R

```diff
@@ -0,0 +1,143 @@
+    codecov_url <- paste0(base_url, "/upload/v2?service=circleci",
+                          "&branch=", Sys.getenv("CIRCLE_BRANCH"),
+                          "&build=", Sys.getenv("CIRCLE_BUILD_NUM"),
+                          "&owner=", Sys.getenv("CIRCLE_PROJECT_USERNAME"),
+                          "&repo=", Sys.getenv("CIRCLE_PROJECT_REPONAME"),
+                          "&commit=", Sys.getenv("CIRCLE_SHA1"))
+  # ---------
+  # Semaphore
+  # ---------
+  } else if (Sys.getenv("CI") == "true" && Sys.getenv("SEMAPHORE") == "true") {
+    # https://semaphoreapp.com/docs/available-environment-variables.html
+    codecov_url <- paste0(base_url, "/upload/v2?service=semaphore",
+                          "&branch=", Sys.getenv("BRANCH_NAME"),
+                          "&build=", Sys.getenv("SEMAPHORE_BUILD_NUMBER"),
+                          "&owner=", strsplit(Sys.getenv("SEMAPHORE_REPO_SLUG"), "/")[0],
+                          "&repo=", strsplit(Sys.getenv("SEMAPHORE_REPO_SLUG"), "/")[1],
+                          "&commit=", Sys.getenv("REVISION"))
+  # --------
+  # drone.io
+  # --------
+  } else if (Sys.getenv("CI") == "true" && Sys.getenv("DRONE") == "true") {
+    # http://docs.drone.io/env.html
+    codecov_url <- paste0(base_url, "/upload/v2?service=drone.io",
+                          "&branch=", Sys.getenv("DRONE_BRANCH"),
+                          "&build=", Sys.getenv("DRONE_BUILD_NUMBER"),
+                          "&build_url=", Sys.getenv("DRONE_BUILD_URL"),
+                          "&commit=", Sys.getenv("DRONE_COMMIT"))
+  # --------
+  # AppVeyor
+  # --------
+  } else if (Sys.getenv("CI") == "True" && Sys.getenv("APPVEYOR") == "True") {
+    # http://www.appveyor.com/docs/environment-variables
+    codecov_url <- paste0(base_url, "/upload/v2?service=AppVeyor",
+                          "&branch=", Sys.getenv("APPVEYOR_REPO_BRANCH"),
+                          "&build=", Sys.getenv("APPVEYOR_BUILD_NUMBER"),
+                          "&owner=", strsplit(Sys.getenv("APPVEYOR_REPO_NAME"), "/")[0],
+                          "&repo=", strsplit(Sys.getenv("APPVEYOR_REPO_NAME"), "/")[1],
+                          "&commit=", Sys.getenv("APPVEYOR_REPO_COMMIT"))
+  # -------
+  # Wercker
+  # -------
+  } else if (Sys.getenv("CI") == "true" && Sys.getenv("WERCKER_GIT_BRANCH")) {
+    # http://devcenter.wercker.com/articles/steps/variables.html
+    codecov_url <- paste0(base_url, "/upload/v2?service=werker",
+                          "&branch=", Sys.getenv("WERCKER_GIT_BRANCH"),
+                          "&build=", Sys.getenv("WERCKER_MAIN_PIPELINE_STARTED"),
+                          "&owner=", Sys.getenv("WERCKER_GIT_OWNER"),
+                          "&repo=", Sys.getenv("WERCKER_GIT_REPOSITORY"),
+                          "&commit=", Sys.getenv("WERCKER_GIT_COMMIT"))
+  # ---------
+  # Local GIT
+  # ---------
+  } else {
+    branch <- local_branch()
+    codecov_url <- paste0(base_url, "/upload/v2",
```

Comment:
This returned a `Method Not Allowed` (405) error when I tried to upload (L119) the below report:

```r
> coverage
{"files":[{"name":"/home/daroczig/projects/r-test//R/r-test.R","coverage":[null,null,null,null,null,2,2,2,0,2]}],"uploader":["R"]} 
```